File: consumer.js
var amqp = require('amqplib/callback_api');
var total = 0;
var loops = 10;

function init() {
    amqp.connect("amqp://guest:guest@192.168.99.100:5672", function(err, conn) {
        conn.createChannel(function(err, ch) {
            ch.prefetch(1);
            test(ch);
        });
    }); 
}

function test(ch) {
    ch.consume('speed_test', function(msg) {
        var body = JSON.parse(msg.content.toString());
        var value = 0;
        for(var i = 0; i < body.value; i++)
            value += i;
        total++;
        ch.ack(msg);
    }, {noAck: false});
}

var stats = [];
function stat() {
    var _t = total;
    var dur = new Date() - start;
    stats.push(_t / (dur/1000));
    loops--;
    console.log("Processed " + _t + " in " + (dur/1000) +  " seconds.")
    console.log(loops + " Loops left...");
    if(loops > 0)
        setTimeout(stat, 10000);
    else
        finalize();
    start = new Date();
    total = 0;
}

function finalize() {
    var t = 0;
    for(var i in stats)
        t += stats[i]
    console.log("Average Transfer Rate: " + t/stats.length );
}

var timeout = new Date();
var start = new Date();
timeout.setSeconds(timeout.getSeconds() + 10);
init();
setTimeout(stat, 10000);

File: consumer.py
from time import time
import json
import pika

total = 0

params = pika.URLParameters("amqp://guest:guest@192.168.99.100:5672")
rabbit = pika.BlockingConnection(params)
channel = rabbit.channel()
# channel.queue_declare(queue='speed_test', durable=False)

def handler(ch, meth, prop, body):
    global total
    _body = json.loads(body)
    value = 0;
    for i in range(_body['value']):
        value += i
    ch.basic_ack(delivery_tag=meth.delivery_tag)
    total += 1

channel.basic_qos(prefetch_count=1)
channel.basic_consume(handler, queue='speed_test');

def test(r):
    global total
    total = 0
    timeout = time() + 10
    _start = time()
    while time() < timeout:
        r.process_data_events()
    _end = time()
    _dur = _end - _start
    return total, _dur, total / _dur


if __name__ == "__main__":
    loops = 10
    vals = []
    while loops > 0:
        c, t, r = test(rabbit)
        vals.append(r);
        loops -= 1;
        print "Processed {} in {} seconds.".format(c, t)
        print "{} loops left...".format(loops);
    print "Average Transfer Rate: {}".format(sum(vals) / len(vals))
